jvm/src/test/scala/tastyquery/TypeSuite.scala

```diff
@@ -575,4 +589,77 @@ class TypeSuite extends UnrestrictedUnpicklingSuite {
     assertIsSymbolWithPath(getX)(getXRef.resolveToSymbol)
   }
 
+  testWithContext("select-field-from-generic-class") {
+    val GenClass = resolve(name"simple_trees" / tname"GenericClass").asClass
+    val PolySelect = resolve(name"simple_trees" / tname"PolySelect").asClass
+    val IntClass = resolve(name"scala" / tname"Int")
+
+    val Some(DefDef(_, _, _, body, _)) = PolySelect.lookup(name"testField").get.tree: @unchecked
+
+    val Select(qual, fieldName) = body: @unchecked
+
+    assert(clue(qual.tpe).isApplied(_.isOfClass(GenClass), List(_.isOfClass(IntClass))))
+    assertEquals(fieldName, name"field")
+    assert(clue(body.tpe).isOfClass(IntClass))
+  }
+
+  testWithContext("select-getter-from-generic-class") {
+    val GenClass = resolve(name"simple_trees" / tname"GenericClass").asClass
+    val PolySelect = resolve(name"simple_trees" / tname"PolySelect").asClass
+    val IntClass = resolve(name"scala" / tname"Int")
+
+    val Some(DefDef(_, _, _, body, _)) = PolySelect.lookup(name"testGetter").get.tree: @unchecked
+
+    val Select(qual, getterName) = body: @unchecked
+
+    assert(clue(qual.tpe).isApplied(_.isOfClass(GenClass), List(_.isOfClass(IntClass))))
+    assertEquals(getterName, name"getter")
+    body.tpe
```

Comment:
is this line necessary? it is evaluated again in the next line

Reply:
ok test behaves the same if I delete it

Reply:
That would be leftover from debugging. ;)